inspectit-oce-core/src/test/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContextUnitTest.java

```diff
@@ -0,0 +1,748 @@
+            root.makeActive(false);
+            root.setData("tag", "invisibleValue");
+
+            AtomicReference<Object> tagValue = new AtomicReference<>();
+            Runnable asyncTask = Context.current().wrap(() -> {
+                InspectitContext asyncChild = InspectitContext.createFromCurrent(commonTags, propagation, false);
+                tagValue.set(asyncChild.getData("tag"));
+                asyncChild.makeActive(false);
+                asyncChild.close();
+            });
+
+            root.close();
+
+            asyncTask.run();
+
+            assertThat(tagValue.get()).isEqualTo("rootValue");
+        }
+
+    }
+
+
+    @Nested
+    public class UpPropagation {
+
+        @Test
+        void verifyNewValuesPropagatedWhenConfigured() {
+            when(commonTags.getCommonTagValueMap()).thenReturn(Collections.emptyMap());
+            doReturn(true).when(propagation).isPropagatedUpWithinJVM(eq("tag1"));
+            doReturn(false).when(propagation).isPropagatedUpWithinJVM(eq("tag2"));
+
+            InspectitContext ctxA = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            ctxA.makeActive(true);
+            InspectitContext ctxB = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            ctxB.makeActive(true);
+            InspectitContext ctxC = InspectitContext.createFromCurrent(commonTags, propagation, true);
+            ctxC.makeActive(true);
+
+            ctxC.setData("tag1", "ctxC_value1");
+            ctxC.setData("tag2", "ctxC_value2");
+
+            assertThat(ctxA.getData("tag1")).isNull();
+            assertThat(ctxA.getData("tag2")).isNull();
+
+            assertThat(ctxB.getData("tag1")).isNull();
+            assertThat(ctxB.getData("tag2")).isNull();
+
+            assertThat(ctxC.getData("tag1")).isEqualTo("ctxC_value1");
+            assertThat(ctxC.getData("tag2")).isEqualTo("ctxC_value2");
+
+            ctxC.close();
+
+            assertThat(ctxA.getData("tag1")).isNull();
+            assertThat(ctxA.getData("tag2")).isNull();
+
+            assertThat(ctxB.getData("tag1")).isEqualTo("ctxC_value1");
+            assertThat(ctxB.getData("tag2")).isNull();
+
+            ctxB.close();
+
+            assertThat(ctxA.getData("tag1")).isEqualTo("ctxC_value1");
+            assertThat(ctxA.getData("tag2")).isNull();
+
+            ctxA.close();
+
+            assertThat(InspectitContext.INSPECTIT_KEY.get()).isNull();
+        }
+
+
+        @Test
+        void verifyNoUpPropagationForAsyncChildrenOnDifferentThreadWithRootActive() throws Exception {
+            when(commonTags.getCommonTagValueMap()).thenReturn(Collections.emptyMap());
+            lenient().doReturn(true).when(propagation).isPropagatedUpWithinJVM(any());
+
+            InspectitContext root = InspectitContext.createFromCurrent(commonTags, propagation, false);
+            root.setData("tag", "rootValue");
+            root.makeActive(false);
```

Comment:
why are you always calling `false` when testing "async" stuff, this is only related to the OC context, shouldn't `true` here also work fine..